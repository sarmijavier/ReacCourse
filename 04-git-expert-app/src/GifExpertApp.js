import React, { Fragment , useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GifExpertApp = ({ defaultCategories = [] }) => {
    
    
    const [categories, setCategories] = useState( defaultCategories );
    
    
    return (
        <Fragment>
            <h2>Gif Expert App</h2>

            <AddCategory setCategories={ setCategories }/>

            <hr />

            <ol>
                { 
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </Fragment>
    )
}


export default GifExpertApp;