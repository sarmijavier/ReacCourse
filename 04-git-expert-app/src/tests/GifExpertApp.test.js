import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
import React from 'react';


describe('Pruebas en <GifExpertApp />', () => {
    
    test('Debe  mostrarse correctamente', () => {
        
        const wrapper = shallow( <GifExpertApp/> );
    
        expect( wrapper ).toMatchSnapshot();
    
    })
    
    
    test('debe de mostrar una lista de cartegorias', () => {
        const categories = ['One punch', 'Dragon ball'];        
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
})

