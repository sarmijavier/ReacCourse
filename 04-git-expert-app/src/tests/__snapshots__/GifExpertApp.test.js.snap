// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe  mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de cartegorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
